hydra-node/src/Hydra/Events/FileBased.hs

```diff
@@ -27,11 +27,14 @@ import Hydra.Persistence (PersistenceIncremental (..))
 --
 -- (ToJSON e, FromJSON e, HasEventId) e => (EventSource e m, EventSink e m)
 eventPairFromPersistenceIncremental ::
-  (IsChainState tx, MonadSTM m) =>
+  (IsChainState tx, MonadSTM m, MonadTime m) =>
   PersistenceIncremental (PersistedStateChange tx) m ->
   m (EventSource (StateEvent tx) m, EventSink (StateEvent tx) m)
 eventPairFromPersistenceIncremental PersistenceIncremental{append, source} = do
   eventIdV <- newTVarIO Nothing
+  -- TODO: All legacy events will have the same associated time. How to work around this?
+  now <- getCurrentTime
```

Comment:
This is not only an issue for legacy events, but also for "new" events as we are _adding_ the `time` field in this PR. In fact, we can and should drop the `Legacy` parsing as we have released (major) versions of the `hydra-node` which required wiping of the persistence dir (double-check please)

Should: get current time and add it while parsing (move `getCurrentTime` to conduit)
Should: drop the `Legacy` mode (and mention it in CHANGELOG)